lib/IRGen/GenDecl.cpp

```diff
@@ -832,7 +832,8 @@ void IRGenerator::emitGlobalTopLevel() {
   bool isWholeModule = PrimaryIGM->getSILModule().isWholeModule();
   for (SILFunction &f : PrimaryIGM->getSILModule()) {
     // Only eagerly emit functions that are externally visible.
-    if (!isPossiblyUsedExternally(f.getLinkage(), isWholeModule))
+    if (!isPossiblyUsedExternally(f.getLinkage(), isWholeModule) &&
+        !f.hasSemanticsAttr("stdlib_binary_only"))
```

Comment:
Note that once we get rid of -sil-serialize-all, stdlib_binary_only will go away, because it will be the same as "function not annotated @_inlineable"